underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystemFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.s3;
+
+import java.io.IOException;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.conf.TachyonConf;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.underfs.UnderFileSystemFactory;
+
+public class S3UnderFileSystemFactory implements UnderFileSystemFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  @Override
+  public UnderFileSystem create(String path, TachyonConf tachyonConf, Object conf) {
+    Preconditions.checkArgument(path != null, "path may not be null");
+
+    if (addAndCheckAWSCredentials(tachyonConf)) {
+      TachyonURI uri = new TachyonURI(path);
+      return new S3UnderFileSystem(uri.getHost(), tachyonConf);
+    } else {
+      String err = "AWS Credentials not available, cannot create S3 Under File System.";
+      LOG.error(err);
+      throw Throwables.propagate(new IOException(err));
+    }
+  }
+
+  @Override
+  public boolean supportsPath(String path, TachyonConf tachyonConf) {
+    return path != null && path.startsWith(Constants.HEADER_S3N);
+  }
+
+  private boolean addAndCheckAWSCredentials(TachyonConf tachyonConf) {
+    String accessKeyConf = "fs.s3n.awsAccessKeyId";
```

Comment:
Actually, just realized that both const strings "fs.s3n.awsSecretAccessKey" and "fs.s3n.awsAccessKeyId" appear multiple times in this package. It make better sense to have them either in tachyon.Constants, or define them in something like tachyon.underfs.s3.Constants.java, and refer them using the constant variable name rather than using inline strings. In that way, mis-spelling in the string content will be caught by compiler/IDE before runtime.